snapshots/manager.go

```diff
@@ -257,3 +290,44 @@ func (m *Manager) RestoreChunk(chunk []byte) (bool, error) {
 	}
 	return false, nil
 }
+
+// Snapshot takes a snapshot of the current state and prunes any old snapshottypes.
+func (m *Manager) Snapshot(height int64) {
```

Comment:
The name feels a bit misleading in that its only doing a snapshot if its on a real snapshot height. (The name feels like it'd always force a snapshot.

wdyt of calling it "SnapshotIfApplicable", or something of that regard? If not, perhaps we can update the comment to 

```suggestion
// Snapshot takes a snapshot of the current state if we are on a snapshot height. It also prunes any old snapshots. The snapshotting and pruning happens in a separate process.
func (m *Manager) Snapshot(height int64) {
```

Reply:
Makes perfect sense, will get that changed